node_modules/oae-content/lib/search.js

```diff
@@ -166,76 +166,77 @@ ContentAPI.on(ContentConstants.events.DELETED_COMMENT, function(ctx, comment, co
     // If there were no messages stored on the resource object, we go ahead and index all comments for the content item
-    MessageBoxSearch.createAllMessageSearchDocuments(ContentConstants.search.MAPPING_CONTENT_COMMENT, resource.id, resource.id, function(err, documents) {
-        if (err) {
-            log().warn({'err': err, 'resource': resource}, 'An error occurred producing message search documents');
+    MessageBoxSearch.createAllMessageSearchDocuments(ContentConstants.search.MAPPING_CONTENT_COMMENT, resource.id, resource.id, function(errors, documents) {
+        if (!_.isEmpty(errors)) {
+            _errors = _.union(_errors, errors);
         }
 
         _documents = _.union(_documents, documents);
-        return _produceContentCommentDocuments(resources, callback, _documents);
+        return _produceContentCommentDocuments(resources, callback, _documents, _errors);
     });
 };
 
 /**
  * Produce the necessary content body search documents.
  *
- * @param  {Object[]}   resources           The resources representing the content item whose body to index
- * @param  {Function}   callback            Standard callback method
- * @param  {Object}     callback.err        An error that occurred, if any
- * @param  {Object[]}   callback.documents  The content body search documents
+ * @see SearchAPI#registerChildSearchDocument
  * @api private
  */
-var _produceContentBodyDocuments = function(resources, callback, _documents) {
+var _produceContentBodyDocuments = function(resources, callback, _documents, _errors) {
     _documents = _documents || [];
+    _errors = _errors || [];
     if (_.isEmpty(resources)) {
-        return callback(null, _documents);
+        return callback(_errors, _documents);
     }
 
     var resource = resources.pop();
     // Get the latest revision
     ContentDAO.Revisions.getRevisions(resource.id, null, 1, null, function(err, revisions) {
         if (err) {
-            return callback(err);
+            _errors.push(err);
+            return _produceContentBodyDocuments(resources, callback, _documents, _errors);
         }
+
         var revision = revisions[0];
         var body = [];
 
         // Skip revisions that don't have (html) previews
-        if (!revision.previews || !revision.previews.pageCount) {
+        if (!revision.previews || revision.previews.status !== ContentConstants.previews.DONE || !revision.previews.pageCount) {
             log().trace({'id': resource.id, 'previews': revision.previews}, 'No text to index');
-            return callback(null, _documents);
-        }
 
-        var tenantAlias = revision.previewsId.split(':')[1];
+            // Move on to the next resource
+            return _produceContentBodyDocuments(resources, callback, _documents, _errors);
+        }
 
         ContentDAO.Previews.getContentPreview(revision.previewsId, 'plain.txt', function(err, preview) {
             if (err) {
-                return callback(err);
+                _errors.push(err);
```

Comment:
This could be `_errs = _.union(_errs, [err]);` if we got the route of leaving `_errs` as undefined until an error happens